src/main/webapp/script.js

```diff
@@ -0,0 +1,33 @@
+function ratePromoCode(tableRow, rating) {
+    var videoUrl = document.getElementById('promoCodeTable').rows[tableRow].cells[0].firstChild.getAttribute("href");
+    fetch('/promo-code?rating=' + rating + '&videoUrl=' + videoUrl, {method: 'POST'});
+    displayCodes();
+}
+
+async function displayCodes() {
+    const response = await fetch('/promo-code');
+    const codes = await response.json();
+    const tableEl = document.getElementById('promoCodeTable');
+    tableEl.innerHTML =
+                `<tr id="row0">
+                    <th>VideoURL</th>
+                    <th>PROMO CODE</th>
+                    <th>Rating</th>
+                    <th><button>Like</button></th>
+                    <th><button>Dislike</button></th>
+                </tr>`;
+    const numOfCodes = Object.keys(codes).length;
+    for (i = 0; i < numOfCodes; i++) {
+        var row = tableEl.insertRow(-1);
+        var videoUrl = row.insertCell(0);
+        var promoCode = row.insertCell(1);
+        var rating = row.insertCell(2);
+        var likeButton = row.insertCell(3);
```

Comment:
iirc these likely won't be needed anymore according to the rescope, but feel free to iterate on that in a subsequent cl.